lib/experimental/site-editor-permalinks.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+add_action(
+	'block_editor_settings_all',
+	function ( $settings ) {
+		$settings['__experimentalDashboardLink'] = admin_url();
+		return $settings;
+	}
+);
+
+function gutenberg_remove_query_args( $args ) {
+	$query_string = $_SERVER['QUERY_STRING'];
+	$query        = wp_parse_args( $query_string );
+	foreach ( $args as $arg_name ) {
+		unset( $query[ $arg_name ] );
+	}
+	return $query;
+}
+
+function gutenberg_get_site_editor_url( $path = '', $query = array() ) {
+	$query_string = build_query( array_merge( $query, $path ? array( 'p' => $path ) : array() ) );
```

Comment:
Also, it could make sense to not call `build_query` at all if there are no args to build it from.